OpenEdXMobile/src/main/java/org/edx/mobile/view/AuthenticatedWebViewFragment.java

```diff
@@ -0,0 +1,359 @@
+        } else {
+            hideLoadingProgress();
+        }
+    }
+
+    /**
+     * Shows the error message with the given icon, if the web page failed to load
+     *
+     * @param errorMsg  The error message to show
+     * @param errorIcon The error icon to show with the error message
+     */
+    private void showErrorMessage(@StringRes int errorMsg, @NonNull Icon errorIcon) {
+        if (!pageIsLoaded) {
+            tryToClearWebView();
+            showErrorView(getString(errorMsg), errorIcon);
+        }
+    }
+
+    private void showErrorView(@NonNull String errorMsg, @NonNull Icon errorIcon) {
+        final Context context = getContext();
+        errorTextView.setVisibility(View.VISIBLE);
+        errorTextView.setText(errorMsg);
+        errorTextView.setCompoundDrawablesWithIntrinsicBounds(null,
+                new IconDrawable(context, errorIcon)
+                        .sizeRes(context, R.dimen.content_unavailable_error_icon_size)
+                        .colorRes(context, R.color.edx_brand_gray_back),
+                null, null
+        );
+    }
+
+    /**
+     * Hides the error message view and reloads the web page if it wasn't already loaded
+     */
+    private void hideErrorMessage() {
+        errorTextView.setVisibility(View.GONE);
+        if (!pageIsLoaded || didReceiveError) {
+            tryToLoadWebView(true);
+        }
+    }
+
+    private void tryToLoadWebView(boolean forceLoad) {
+        System.gc(); //there is a well known Webview Memory Issue With Galaxy S3 With 4.3 Update
+
+        if ((!forceLoad && pageIsLoaded) || progressWheel == null) {
+            return;
+        }
+
+        if (!EventBus.getDefault().isRegistered(this)) {
+            EventBus.getDefault().register(this);
+        }
+
+        if (!NetworkUtil.isConnected(getContext())) {
+            showErrorMessage(R.string.reset_no_network_message, FontAwesomeIcons.fa_wifi);
+            return;
+        }
+
+        showLoadingProgress();
+
+        if (!TextUtils.isEmpty(url)) {
+            Map<String, String> map = new HashMap<>();
+            final String token = loginPrefs.getAuthorizationHeader();
+            if (token != null) {
+                map.put("Authorization", token);
+            }
+
+            // Requery the session cookie if unavailable or expired.
+            final EdxCookieManager cookieManager = EdxCookieManager.getSharedInstance(getContext());
+            if (cookieManager.isSessionCookieMissingOrExpired()) {
+                cookieManager.tryToRefreshSessionCookie();
+            } else {
+                didReceiveError = false;
+                webView.loadUrl(url);
+            }
+        }
+    }
+
+    private void tryToClearWebView() {
+        pageIsLoaded = false;
+        if (webView != null) {
+            webView.loadData(EMPTY_HTML, "text/html", "UTF-8");
+        }
+    }
+
+    @Override
+    public void setUserVisibleHint(boolean isVisibleToUser) {
```

Comment:
This function is mostly only applicable when we use this fragment in a ViewPager.